Analysis/Tasks/PWGDQ/tableReader.cxx

```diff
@@ -276,32 +262,174 @@ struct MuonTrackSelection {
+    fHistMan->SetUseDefaultVariableNames(kTRUE);
+    fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
+
+    TString histNames = "";
+    TString configCutNamesStr = fConfigElectronCuts.value;
+    if (!configCutNamesStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(configCutNamesStr.Tokenize(","));
+      for (int icut = 0; icut < objArray->GetEntries(); ++icut) {
+        fCutNames.push_back(objArray->At(icut)->GetName());
+        histNames += Form("PairsBarrelMEPM_%s;PairsBarrelMEPP_%s;PairsBarrelMEMM_%s;", fCutNames[icut].Data(), fCutNames[icut].Data(), fCutNames[icut].Data());
+        fTwoTrackFilterMask |= (uint8_t(1) << icut);
       }
     }
+    histNames += "PairsMuonMEPM;PairsMuonMEPP;PairsMuonMEMM;";
+
+    DefineHistograms(fHistMan, histNames.Data());    // define all histograms
+    VarManager::SetUseVars(fHistMan->GetUsedVars()); // provide the list of required variables so that VarManager knows what to fill
+    fOutputList.setObject(fHistMan->GetMainHistogramList());
   }
+
+  void process(soa::Filtered<MyEventsHashSelected>& events, soa::Filtered<MyBarrelTracksSelected> const& tracks, soa::Filtered<MyMuonTracksSelected> const& muons)
+  {
+    uint8_t twoTrackFilter = 0;
+
+    events.bindExternalIndices(&tracks);
+    events.bindExternalIndices(&muons);
+    auto tracksTuple = std::make_tuple(tracks);
+    auto muonsTuple = std::make_tuple(muons);
+    AnalysisDataProcessorBuilder::GroupSlicer slicerTracks(events, tracksTuple);
+    AnalysisDataProcessorBuilder::GroupSlicer slicerMuons(events, muonsTuple);
+
+    // Strictly upper categorised collisions, for 100 combinations per bin, skipping those in entry -1
+    for (auto& [event1, event2] : selfCombinations("fMixingHash", 100, -1, events, events)) {
+
+      // event informaiton is required to fill histograms where both event and pair information is required (e.g. inv.mass vs centrality)
+      VarManager::ResetValues(0, VarManager::kNVars, fValues);
+      VarManager::FillEvent<gkEventFillMap>(event1, fValues);
+
+      auto it1 = slicerTracks.begin();
+      auto it2 = slicerTracks.begin();
+      for (auto& slice : slicerTracks) {
+        if (slice.groupingElement().index() == event1.index()) {
+          it1 = slice;
+          break;
+        }
+      }
+      for (auto& slice : slicerTracks) {
+        if (slice.groupingElement().index() == event2.index()) {
+          it2 = slice;
+          break;
+        }
+      }
+
+      auto tracks1 = std::get<soa::Filtered<MyBarrelTracksSelected>>(it1.associatedTables());
+      tracks1.bindExternalIndices(&events);
+      auto tracks2 = std::get<soa::Filtered<MyBarrelTracksSelected>>(it2.associatedTables());
+      tracks2.bindExternalIndices(&events);
+
+      for (auto& track1 : tracks1) {
+        for (auto& track2 : tracks2) {
+          twoTrackFilter = track1.isBarrelSelected() & track2.isBarrelSelected() & fTwoTrackFilterMask;
```

Comment:
This cannot be done as a filter?

Reply:
As we discussed in mattermost, the bitwise AND for the fTwoTrackFilterMask could in principle be applied as a Filter, although at the moment there are two problems with this:
1) the bitwise AND is not supported in Filter expressions, so as soon as it will be possible, the code will actually change in few other places where i had to apply workarounds due to this missing feature
2) the fTwoTrackFilterMask is computed at init() time because it needs to know how many electron cuts were configured by the user; and this is available via a Configurable